dot/state/block.go

```diff
@@ -153,6 +159,29 @@ func NewBlockStateFromGenesis(db database.Database, trs *Tries, header *types.He
 	return bs, nil
 }
 
+// Pause pauses the service ie. halts block production
+func (bs *BlockState) Pause() error {
+	bs.Lock()
+	defer bs.Unlock()
+
+	if bs.IsPaused() {
+		return nil
+	}
+
+	close(bs.pause)
+	return nil
```

Comment:
If we are using the lock only for this, why if we change this logic to:

```suggestion
        select {
	case <-pause:
	default:
		close(pause)
	}
	return nil
```